code_challenges.rb

```diff
@@ -30,11 +61,57 @@
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+# create a class called Bike
+class Bike 
+# initialize model wheels and speed
+    def initialize(model, wheels, current_speed)
+        @model = model
+        @wheels = wheels
+        @current_speed = 0
+    end
+# create a methode to get info about bike
+    def get_info 
+        "#{@model}, #{@wheels}, #{@current_speed}"
+    end
+end
 
-
+new_bike = Bike.new('Haro', 2, nil)
+p new_bike.get_info
 
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed. The brake method should decrease the speed. The bike cannot go negative speeds.
 
+# create a class called Bike
+class Bike 
+# initialize model wheels speed set to 0
+    def initialize(model, wheels, current_speed)
+        @model = model
+        @wheels = wheels
+        @current_speed = 0
+    end
+# create method to increase speed by 1
+    def pedal_faster
+        @current_speed = @current_speed+1
+    end
+# create method to decrease speed by 0
+    def brake
+# conditional if speed is greater than 0 return speed -1 if not return 0
+        if @current_speed>0
+        return @current_speed = @current_speed-1
+        else
+            return 0
+        end
+    end
+    def get_info 
+        "#{@model}, #{@wheels}, #{@current_speed}"
+    end
+end
+
+new_bike = Bike.new('Haro', 2, nil)
+
+p new_bike.brake
```

Comment:
did you test other methods here? looks good!

Reply:
yea it worked for me